src/hashcat.c

```diff
@@ -2895,7 +2896,7 @@ static int run_kernel (const uint kern_run, hc_device_param_t *device_param, con
     {
       if (data.opti_type & OPTI_TYPE_SLOW_HASH_SIMD)
       {
-        num_elements = CEIL ((float) num_elements / device_param->vector_width);
+        num_elements = CEIL (num_elements / device_param->vector_width);
```

Comment:
Hence the need for the typecast you removed. I'm not saying you are wrong, I'm just asking if you actually understand what you are doing here.